pkg/cluster/prepare.go

```diff
@@ -0,0 +1,130 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/fluxcd/pkg/untar"
+	"github.com/weaveworks/pctl/pkg/runner"
+)
+
+const (
+	kubectlCmd = "kubectl"
+)
+
+// Fetcher will download a manifest tar file from a remote repository.
+type Fetcher struct {
+	Client *http.Client
+}
+
+// Applier applies the previously generated manifest files.
+type Applier struct {
+	Runner runner.Runner
+}
+
+// Preparer will prepare an environment.
+type Preparer struct {
+	PrepConfig
+	Fetcher *Fetcher
+	Applier *Applier
+}
+
+// PrepConfig defines configuration options for prepare.
+type PrepConfig struct {
+	// BaseURL is given even one would like to download manifests from a fork
+	// or a test repo.
+	BaseURL     string
+	Location    string
+	Version     string
+	KubeContext string
+	KubeConfig  string
+	DryRun      bool
+}
+
+// NewPreparer creates a preparer with set dependencies ready to be used.
+func NewPreparer(cfg PrepConfig) (*Preparer, error) {
+	tmp, err := ioutil.TempDir("", "pctl-manifests")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temp folder for manifest files: %w", err)
+	}
+	cfg.Location = tmp
+	return &Preparer{
+		PrepConfig: cfg,
+		Fetcher: &Fetcher{
+			Client: http.DefaultClient,
+		},
+		Applier: &Applier{
+			Runner: &runner.CLIRunner{},
+		},
+	}, nil
+}
+
+// Prepare will prepare an environment with everything that is needed to run profiles.
+func (p *Preparer) Prepare() error {
+	if _, err := os.Stat(p.Location); err != nil {
+		return err
+	}
+	if err := p.Fetcher.Fetch(context.Background(), p.BaseURL, p.Version, p.Location); err != nil {
+		return err
+	}
+	return p.Applier.Apply(p.Location, p.KubeContext, p.KubeConfig, p.DryRun)
+}
+
+// Fetch the latest or a version of the released manifest files for profiles.
+func (f *Fetcher) Fetch(ctx context.Context, url, version, dir string) error {
+	ghURL := fmt.Sprintf("%s/latest/download/manifests.tar.gz", url)
```

Comment:
Even though we allow a base url change so to test forks, this has to match though... And I'm okay with that part. Later we can add doing this with manual resources or something if downloading these manifest files is not something that can be done in an isolated environment from the network.